### PR TITLE
Add location module information.

### PR DESCRIPTION
**Pull request by bryanlogan**
# Pull Request/Issue Resolution

#### Description of Change:
Adding information about the Braze location module for Android

Closes #SDK-3299

#### Is this change associated with a Braze feature/product release?
- [X ] Yes Android SDK 24.0.0, released today.
- [ ] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [ ] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [ ] Check that all links work.
- [ ] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @Timothy-Kim and @KellieHawks as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [ ] Tag others as reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @timothy-kim or @KellieHawks for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs</td>
  </tr>
  <tr>
    <td>@kelliehawks</td>
    <td>Currents<br>Internal Tools<br>Product Partnerships<br>SMS<br>Customer Lifecycle, Identity and Permissions</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Reporting<br>Intelligence<br>User Targeting<br>IAM<br>Channels<br>FIX</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging and Automation<br>Email (Composition and Infrastructure)</td>
  </tr>
</table>
</details>

### CODE REVIEW

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 1]
```diff
@@ -22,10 +22,11 @@ To support geofences for Android:
 
 ## Step 1: Update build.gradle
 
-Add the Google Play Services [location package][3] to your app-level `build.gradle` using the Google Play Services [setup guide][10]:
+Add `android-sdk-location` to your app-level `build.gradle` if using Android SDK version `24.0.0` or greater. Also add the Google Play Services [location package][3] using the Google Play Services [setup guide][10]:
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:25]: I wouldn't include the SDK version here. Usually the docs represent the "current" state of the SDK only, so being on version 24.0.0, 24.0.1, 26.0.0, etc. would be implied.

File: _docs/_developer_guide/platform_integration_guides/android/initial_sdk_setup/android_sdk_integration.md [hunk 2]
```diff
@@ -48,13 +48,16 @@ Alternatively, you can directly find the artifact AAR files on our [maven reposi
 
 Add the `android-sdk-ui` dependency to your app's `build.gradle`. 
 
+If you are using any of the location or geofence functionality and also using and SDK version of `24.0.0` or greater, also include `android-sdk-location` to your app's `build.gradle`.
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/android/initial_sdk_setup/android_sdk_integration.md:51]:
```suggestion
If you are using any location or Braze Geofence functionality, also include `android-sdk-location` to your app's `build.gradle`.
```

**Review by radixdev — APPROVED** [review]

**Comment by radixdev** [comment]
thanks for running with this so quickly 💯

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md (hunk 1 above)
Comment (bryanlogan) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:25]: I saw a few things indicating "If less than...", but I guess for "greater", it makes more sense to leave it implied.

File: _docs/_developer_guide/platform_integration_guides/android/initial_sdk_setup/android_sdk_integration.md [hunk 3]
```diff
@@ -48,13 +48,16 @@ Alternatively, you can directly find the artifact AAR files on our [maven reposi
 
 Add the `android-sdk-ui` dependency to your app's `build.gradle`. 
 
+If you are using any location or Braze Geofence functionality, also include `android-sdk-location` to your app's `build.gradle`.
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/android/initial_sdk_setup/android_sdk_integration.md:51]:
```suggestion
If you are using any location or Braze Geofence functionality, also include `android-sdk-location` in your app's `build.gradle`.
```

**Review by radixdev — APPROVED** [review]

**Review by josh-mccrowell-braze — APPROVED** [review]
Looks good to me! Stripping out appboy must have been a whole bunch of work. Kudos!

**Pull request merged**